pkg/node/manager.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"sync"
+)
+
+var (
+	mutex sync.RWMutex
+	nodes = map[NodeIdentity]*Node{}
+)
+
+func GetNode(ni NodeIdentity) *Node {
+	mutex.RLock()
+	n := nodes[ni]
+	mutex.RUnlock()
+	return n
+}
+
+func UpdateNode(ni NodeIdentity, n *Node) {
+	mutex.Lock()
+	//if oldNode, ok := nodes[ni]; ok {
```

Comment:
The node name can never change so we can never run into a situation where we have to remove on update.